test/mortgage_e2e.ipynb

```diff
@@ -0,0 +1,1123 @@
+    "part_count = 4\n",
+    "\n",
+    "gpu_dfs = []\n",
+    "gpu_time = 0\n",
+    "quarter = 1\n",
+    "year = start_year\n",
+    "count = 0\n",
+    "while year <= end_year:\n",
+    "    for file in glob(os.path.join(perf_data_path + \"/Performance_\" + str(year) + \"Q\" + str(quarter) + \"*\")):\n",
+    "        gpu_dfs.append(process_quarter_gpu(year=year, quarter=quarter, perf_file=file))\n",
+    "        count += 1\n",
+    "    quarter += 1\n",
+    "    if quarter == 5:\n",
+    "        year += 1\n",
+    "        quarter = 1\n",
+    "print(\"ETL for start_year:{} and end_year:{}\\n\".format(start_year,end_year))\n",
+    "wait(gpu_dfs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'tcp://127.0.0.1:33433': None, 'tcp://127.0.0.1:40907': None}"
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "client.run(initialize_rmm_no_pool)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Machine Learning"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Concatonate all the dask dataframes into one.  We will `.persist()` and `wait` here as a milestone to reduce the persist/wait times later."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ddfs = dask_cudf.concat(gpu_dfs)\n",
+    "_ = wait(ddfs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Choose your operation.\n",
+    "You can choose which operation you want to do: **Train on the whole dataset** for use in your OR **Train and test on the dataset** and see an RMSE score"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train on the whole dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we create our data and labels from our dataframe.  We're going to use the `delinquency_12` column as our labels, and the rest of the dataframe as our data.  We have to wait for these to persist in order to use them for the DaskDMatrix, or you will throw an error.  Afterwards, we will train the data and create a model. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X, y = ddfs[ddfs.columns.difference([\"delinquency_12\"])],ddfs[\"delinquency_12\"]\n",
+    "X = X.persist()\n",
+    "_ = wait(X)\n",
+    "y = y.persist()\n",
+    "_ = wait(y) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dtrain = xgb.dask.DaskDMatrix(client, \n",
+    "        X,\n",
+    "        y    )"
```

Comment:
Add `missing=-1` which denote all the nulls in our data (`fillna(-1)`). Should significantly help memory during training and potentially give better results. (? in theory, completely unsure of what actually happens)

Reply:
- [ ] added